akka-stream-testkit/src/main/scala/akka/stream/testkit/StreamTestKit.scala

```diff
@@ -0,0 +1,293 @@
+/**
+ * Copyright (C) 2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.testkit
+
+import scala.language.existentials
+import akka.actor.ActorSystem
+import akka.actor.DeadLetterSuppression
+import akka.stream.impl.{ EmptyPublisher, ErrorPublisher }
+import akka.testkit.TestProbe
+import org.reactivestreams.{ Publisher, Subscriber, Subscription }
+import scala.collection.immutable
+import scala.concurrent.duration._
+import scala.util.control.NoStackTrace
+
+/**
+ * Provides factory methods for various Publishers.
+ */
+object TestPublisher {
+
+  import StreamTestKit._
+
+  /**
+   * Publisher that signals complete to subscribers, after handing a void subscription.
+   */
+  def empty[T](): Publisher[T] = EmptyPublisher[T]
+
+  /**
+   * Publisher that subscribes the subscriber and completes after the first request.
+   */
+  def lazyEmpty[T]: Publisher[T] = new Publisher[T] {
+    override def subscribe(subscriber: Subscriber[_ >: T]): Unit =
+      subscriber.onSubscribe(CompletedSubscription(subscriber))
+  }
+
+  /**
+   * Publisher that signals error to subscribers immediately, before handing out subscription.
+   */
+  def error[T](cause: Throwable): Publisher[T] = ErrorPublisher(cause, "error").asInstanceOf[Publisher[T]]
+
+  /**
+   * Publisher that subscribes the subscriber and signals error after the first request.
+   */
+  def lazyError[T](cause: Throwable): Publisher[T] = new Publisher[T] {
+    override def subscribe(subscriber: Subscriber[_ >: T]): Unit =
+      subscriber.onSubscribe(FailedSubscription(subscriber, cause))
+  }
+
+  /**
+   * Probe that implements [[org.reactivestreams.Publisher]] interface.
+   */
+  def probe[T]()(implicit system: ActorSystem): Probe[T] = new Probe()
+
+  /**
+   * Probe that implements [[org.reactivestreams.Publisher]] interface and tracks demand.
+   */
+  def auto[T](initialPendingRequests: Long = 0)(implicit system: ActorSystem): Auto[T] = new Auto(initialPendingRequests)
```

Comment:
I'm thinking if we should name the `auto` to `probe` instead, to make it the preferred way, and name current `probe` to `manualProbe`